### ISSUE TITLE
HLint suggestions and CPP don't play well together

### ISSUE THREAD

**Issue by LeanderK** [haskell/haskell-ide-engine#675]
When I apply an hlint-suggestion to a piece of code where CPP is used somewhere else in the file, the expanded CPP is weirdly expanded.

**Comment by alanz** [comment]
Yes, the CPP processing is best effort, as it allows pretty much any transformation, which we cannot easily undo to get back to the original source.

It can handle things like `#if` / `#else` etc, for simple conditions. And in the process converts the inactive regions to comments, so will not make any changes within that area.

I would like to see some kind of reduced power preprocessor some time that can only do some basic conditional stuff, which would make this more reliable.

So the rule of thumb now is that using CPP can prevent the tools from working.
